src/controllers/dlgprefcontroller.cpp

```diff
@@ -103,7 +103,11 @@ DlgPrefController::DlgPrefController(
     connect(this,
             &DlgPrefController::applyMapping,
             m_pControllerManager.get(),
-            &ControllerManager::slotApplyMapping);
+            &ControllerManager::slotApplyMapping,
+            Qt::BlockingQueuedConnection);
```

Comment:
This has introduced a deadlock when using with screen controller, which rely on the main thread (GUI) availability ([see doc](https://doc.qt.io/qt-6/qoffscreensurface.html#create))